### ISSUE TITLE
Every depth result we have was measured by a throwaway script — make the reusable half an instrument

### ISSUE THREAD

**Issue by MatthewLacerda2** [MatthewLacerda2/TinyRefinementModel#230]
The referee (`instruments/verdict.py`) and the runner (`instruments/experiment.py`) were
built in #150/#151 to do exactly this. Since then:

```
experiments/*/specs/*.toml   3 specs, all status = "retrofit"
runs/experiments/            does not exist — the runner has never executed
```

**Zero experiments have gone through the machine.** Including the depth work of late
August, which produced the most substantive result the project has: depth recurrence
buys nothing on prose, and on code 76% of the gain lands on 19% of tokens (brackets and
punctuation), scaling with nesting depth. That was measured with throwaway scripts in
gitignored `aux_depth/`. It has no spec, no pre-registered bar, no verdict, and no
findings entry. It exists in a chat log.

## Why it got bypassed — the actual mechanism, not a lapse of discipline

The runner reads `RESULT {...}` lines (`instruments/results.py`). Every ad-hoc probe
printed a human table to stdout instead. Nothing in the repo made the protocol the path
of least resistance, so the path of least resistance won. That is a tooling gap, not a
willpower gap, and it will recur on the next investigation unless the reusable half of
those scripts becomes an instrument that emits `RESULT` by construction.

## What the scratch scripts were, and what each should become

| ad-hoc script | what it did | permanent home |
|---|---|---|
| `paired_depth.py` | per-position CE difference between two depths on identical tokens, with a running mean/variance and a t statistic | `instruments/paired.py` |
| `indomain_depth.py` | pooled CE + argmax disagreement per depth over a corpus | folds into the same instrument as a second readout |
| `why_code.py` | bucketed the CE gain by token category, by confidence decile, by first-occurrence, by bracket nesting | `instruments/attribution.py` |
| `reasoning_tokens.py` | same, bucketed by distance after `=` | a bucketing rule passed to `attribution.py`, not a new file |

### `instruments/paired.py` — the paired-comparison instrument

The statistical point, which I got wrong by hand and want the tool to make unwrong:
comparing one set of weights at two depths on the same tokens is a **paired** test. The
σ≈0.03 nats noise floor from #17 is *seed-to-seed variance across training runs* — the
wrong bar, ~100× too loose, and I reached for it first. The right test is the per-position
difference, whose standard error the tool computes itself.

Requirements:

- Takes two configs (depth, checkpoint, dtype — anything the contract exposes) and a
  corpus; returns mean difference, SE, t, and n.
- **Refuses to run without a control corpus.** Every real conclusion in the depth work
  came from a contrast (code vs prose), and every wrong turn came from a pooled number
  with nothing to compare against. Make the contrast structural.
- Emits `RESULT` lines, so a spec can drive it and the referee can judge it.
- Non-finite guard: #229 showed a checkpoint returning all-NaN logits on ~10% of code
  documents with nothing warning. A measurement instrument must count and report those,
  never average over them.

### `instruments/attribution.py` — where a difference lives

Given a per-position difference and a bucketing rule, report mean gain, share of tokens,
and share of total gain per bucket. This is the tool that turned "depth helps on code"
into "depth tracks nested structure" — it killed three competing explanations in one
run (tokenizer artifact, long-range copy, uncertainty resolution) because share-of-gain
against share-of-tokens falsifies all three at a glance.

Bucketing rules are data, not code: token category, confidence decile, first-occurrence,
nesting depth, distance-after-a-token-set. New question, new rule, same instrument.

## Then re-run the depth investigation through it

Not for a new answer — to prove the loop end to end on a result whose shape is already
known, and to give the finding a home. Concretely: a spec with the delimiter-attribution
hypothesis and a pre-registered bar, the sweep run by `python -m instruments.experiment`,
and whatever the referee says written down. Cross-checkpoint replication across the
0.78B / 1.40B / 3.20B / 3.99B ladder is part of the same spec — a gain that grows with
training is a finding; one that appears only at the end is a curiosity. (The 3.20B rung
will need #229 resolved or explicitly excluded, since it is the NaN checkpoint.)

## Done means

- `instruments/paired.py` and `instruments/attribution.py` exist, emit `RESULT`, and are
  covered in `tests/apparatus/`.
- A spec for the depth-attribution question exists, was committed **before** its results,
  and `runs/experiments/` contains its journal.
- The verdict is recorded — and if it survives, a `docs/findings/` entry replaces the
  chat log.
